cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SubqueryExpressionsTest.scala

```diff
@@ -0,0 +1,250 @@
+        """EXISTS {
+          #  MATCH [Pattern]
+          #  WHERE [Expression]
+          #}""".stripMargin('#'))
+      p("It is worth noting that the `MATCH` keyword can be omitted in such subqueries and that the `WHERE` clause is optional.")
+
+      section("Simple `EXISTS` subquery", "existential-subquery-simple-case") {
+        p(
+          """Variables introduced by the outside scope can be used in the `EXISTS` subquery without importing them
+            #<<subquery-correlated-importing,as necessary with `CALL` subqueries>>.
+            #The following example shows this:""".stripMargin('#'))
+        query(
+          """MATCH (person:Person)
+            #WHERE EXISTS {
+            #  MATCH (person)-[:HAS_DOG]->(:Dog)
+            #}
+            #RETURN person.name AS name""".stripMargin('#'),
+          ResultAssertions(r => {
+            r.toList should equal(List(Map("name" -> "Andy"), Map("name" -> "Peter")))
+          })) {
+          resultTable()
+        }
+      }
+      section("`EXISTS` subquery with `WHERE` clause", "existential-subquery-with-where") {
+        p(
+          """A `WHERE` clause can be used in conjunction to the `MATCH`.
+            #Variables introduced by the `MATCH` clause and the outside scope can be used in this scope.""".stripMargin('#'))
+        query(
+          """MATCH (person:Person)
+            #WHERE EXISTS {
+            #  MATCH (person)-[:HAS_DOG]->(dog:Dog)
+            #  WHERE person.name = dog.name
+            #}
+            #RETURN person.name AS name""".stripMargin('#'),
+          ResultAssertions(r => {
+            r.toList should equal(List(Map("name" -> "Andy")))
+          })) {
+          resultTable()
+        }
+      }
+      section("Nesting `EXISTS` subqueries", "existential-subquery-nesting") {
+        p(
+          """`EXISTS` subqueries can be nested like the following example shows.
+            #The nesting also affects the scopes.
+            #That means that it is possible to access all variables from inside the subquery which are either from the outside scope or defined in the very same subquery.""".stripMargin('#'))
+        query(
+          """MATCH (person:Person)
+            #WHERE EXISTS {
+            #  MATCH (person)-[:HAS_DOG]->(dog:Dog)
+            #  WHERE EXISTS {
+            #    MATCH (dog)-[:HAS_TOY]->(toy:Toy)
+            #    WHERE toy.name = 'Banana'
+            #  }
+            #}
+            #RETURN person.name AS name""".stripMargin('#'),
+          ResultAssertions(r => {
+            r.toList should equal(List(Map("name" -> "Peter")))
+          })) {
+          resultTable()
+        }
+      }
+      section("`EXISTS` subquery outside of a `WHERE` clause", "existential-subquery-outside-where") {
+        p(
+          """`EXISTS` subquery expressions can appear anywhere that an expression is valid.
+            |Here the result if the subquery can find the given pattern is returned.""".stripMargin)
```

Comment:
```suggestion
            |Here is the result in case the subquery can find the given pattern:""".stripMargin)
```

Reply:
No, that is not how I meant it. But I can see it is not clear.
I suggest "Here the result is a boolean that shows whether the subquery can find the given pattern."

Reply:
Yep! I think your suggestion makes it clearer now. Thanks!